oak/src/oak/server_test.cpp

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2020 Robert Bosch GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/**
+ * \file oak/server_test.cpp
+ * \see  oak/server_test.hpp
+ */
+
+#include <gtest/gtest.h>  // for TEST, EXPECT_TRUE, ...
+
+#include <iostream>
+#include <map>      // for map<>
+#include <string>   // for string
+#include <utility>  // for tie
+#include <vector>   // for vector<>
+
+#include <array>
+#include <cstdio>
+#include <iostream>
+#include <memory>
+#include <stdexcept>
+#include <string>
+
+#include <boost/format.hpp>
+
+#include "cloe/registrar.hpp"
+
+#include "oak/registrar.hpp"  // for Registrar
+#include "oak/server.hpp"     // for Server
+
+using namespace std;  // NOLINT(build/namespaces)
+
+/**
+ * Executes a (shell) command and return the StdOut
+ */
+std::string exec(const char* cmd) {
+  std::array<char, 128> buffer;
+  std::string result;
+  std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd, "r"), pclose);
```

Comment:
This is lean... but is this portable? We could also use `boost::process`, which we are using already and have already used before (see `engine/src/command.cpp` I think).